src/cli/cmd/init/mod.rs

```
@@ -0,0 +1,347 @@
+            println!("Exiting. Let's a build a new flake soon, though :)");
+            return Ok(ExitCode::SUCCESS);
+        }
+
+        println!("Let's build a Nix flake!");
+
+        let project = Project::new(self.root);
+        let description = Prompt::maybe_string("An optional description for your flake")?;
+
+        fn get_systems() -> Result<Vec<String>, FhError> {
+            let selected = Prompt::guided_multi_select(
+                "Which systems would you like to support?",
+                "system",
+                SYSTEMS.to_vec(),
+            )?;
+
+            if selected.is_empty() {
+                println!("❌ You need to select at least one system to support");
+                #[allow(clippy::needless_return)]
+                return get_systems();
+            } else {
+                Ok(selected)
+            }
+        }
+
+        let systems = get_systems()?;
+
+        // We could conceivably create a version of `fh init` Nixpkgs included only if certain other choices
+        // are made. But for the time being so much relies on it that we don't have a great opt-out story,
+        // so best to just include it in all flakes.
+        let nixpkgs = match Prompt::select(
+            "Which Nixpkgs version would you like to include?",
+            &[
+                "23.05",
+                "latest",
+                "unstable",
+                "select a specific release (not recommended in most cases)",
+            ],
+        )?
+        .as_str()
+        {
+            // MAYBE: find an enum-based approach to this
+            "23.05" => String::from("0.2305.*"),
+            "latest" => String::from("*"),
+            "unstable" => String::from("0.1.*"),
+            "select a specific release (not recommended in most cases)" => {
+                select_nixpkgs(&self.api_addr).await?
+            }
+            _ => String::from("*"), // Unreachable
```

If it's unreachable, we should throw an error if they somehow reach it.